internal/store/serverservice/attributes.go

```diff
@@ -119,21 +121,48 @@ func (r *Store) deviceVendorData(asset *model.Asset) map[string]string {
 }
 
 // returns a map with device vendor attributes when an update is required
+//
+// nolint:gocyclo // data verify is cyclomatic
 func vendorDataUpdate(newData, currentData map[string]string) map[string]string {
+	const unknown = "unknown"
+	var changes bool
+
 	if currentData == nil {
 		return newData
 	}
 
-	var changes bool
+	newModelNumber := common.FormatProductName(newData[serverModelAttributeKey])
+	currentModelNumber := currentData[serverModelAttributeKey]
 
-	setValue := func(key string, newData, currentData map[string]string) {
-		const unknown = "unknown"
+	changeRequired := func(key string) bool {
+		newModelDefined := newModelNumber != unknown && newModelNumber != ""
+		currentModelUndefined := currentModelNumber == unknown || currentModelNumber == ""
+
+		// model number to be updated
```

Comment:
Why are we even querying things that will never change?

Reply:
for the record, the reasons were discussed in another channel, to summarise,
theres a few exceptional cases like - when bmc-toolbox/common does not yet format the model number the model attribute in the store needs to be updated.